test/integration/test_integration.jl

```diff
@@ -182,49 +226,92 @@ end
         λ = shape_functions(fs, shape(ctype))
         ∇λ = ξ -> ∂λξ_∂x(fs, Val(1), ctype, cnodes, ξ)
         degree = Val(2)
         quad = Quadrature(degree)
-        @test Bcube.integrate_ref(ξ -> λ(ξ)[1] * λ(ξ)[1], ctype, cnodes, quad) ≈ Δx * Δy / 9
-        @test Bcube.integrate_ref(ξ -> λ(ξ)[1] * λ(ξ)[2], ctype, cnodes, quad) ≈
-              Δx * Δy / 18
-        @test Bcube.integrate_ref(ξ -> λ(ξ)[2] * λ(ξ)[2], ctype, cnodes, quad) ≈ Δx * Δy / 9
-        @test Bcube.integrate_ref(ξ -> ∇λ(ξ)[1, :] ⋅ ∇λ(ξ)[1, :], ctype, cnodes, quad) ≈
-              (Δx^2 + Δy^2) / (3 * Δx * Δy)
+        @test integrate_on_ref_element(
+            ReferenceFunction(ξ -> λ(ξ)[1] * λ(ξ)[1]),
+            cInfo,
+            quad,
+        ) ≈ Δx * Δy / 9
+        @test integrate_on_ref_element(
+            ReferenceFunction(ξ -> λ(ξ)[1] * λ(ξ)[2]),
+            cInfo,
+            quad,
+        ) ≈ Δx * Δy / 18
+        @test integrate_on_ref_element(
+            ReferenceFunction(ξ -> λ(ξ)[2] * λ(ξ)[2]),
+            cInfo,
+            quad,
+        ) ≈ Δx * Δy / 9
+        @test integrate_on_ref_element(
+            ReferenceFunction(ξ -> ∇λ(ξ)[1, :] ⋅ ∇λ(ξ)[1, :]),
+            cInfo,
+            quad,
+        ) ≈ (Δx^2 + Δy^2) / (3 * Δx * Δy)
     end
 
     @testset "Line (not curved) - surfacic" begin
         Δx = 666.0
         Δy = 314.0
         mesh = basic_mesh(; coef_x = Δx, coef_y = Δy)
+        c2f = connectivities_indices(mesh, :c2f)
+        f2n = connectivities_indices(mesh, :f2n)
 
         g(x) = 1.0
-        c2n = connectivities_indices(mesh, :c2n)
 
         for idegree in 1:3
             degree = Val(idegree)
             quad = Quadrature(degree)
 
             for icell in (1, 2)
-                cnodes = get_nodes(mesh, c2n[icell])
-                ctype = cells(mesh)[icell]
-                @test Bcube.integrate(g, 1, ctype, cnodes, quad) ≈ Δx
-                @test Bcube.integrate(g, 2, ctype, cnodes, quad) ≈ Δy
-                @test Bcube.integrate(g, 3, ctype, cnodes, quad) ≈ Δx
-                @test Bcube.integrate(g, 4, ctype, cnodes, quad) ≈ Δy
+                # cinfo = CellInfo(mesh, icell)
+                c = CellInfo(mesh, icell)
+                _c2f = c2f[icell]
+
+                kface = _c2f[1]
+                _f2n = f2n[kface]
+                finfo_1 =
+                    FaceInfo(c, c, faces(mesh)[kface], get_nodes(mesh, _f2n), _f2n, kface)
+                kface = _c2f[2]
+                _f2n = f2n[kface]
+                finfo_2 =
+                    FaceInfo(c, c, faces(mesh)[kface], get_nodes(mesh, _f2n), _f2n, kface)
+                kface = _c2f[3]
+                _f2n = f2n[kface]
+                finfo_3 =
+                    FaceInfo(c, c, faces(mesh)[kface], get_nodes(mesh, _f2n), _f2n, kface)
+                kface = _c2f[4]
+                _f2n = f2n[kface]
+                finfo_4 =
+                    FaceInfo(c, c, faces(mesh)[kface], get_nodes(mesh, _f2n), _f2n, kface)
```

Comment:
Can you replace with the following if it works:
```julia
finfo_1, finfo_2, finfo_3, finfo_4 = map( Base.Fix1(FaceInfo, mesh), c2f[icell])
```

Reply:
(and remove the two previous lines)

Reply:
yes I can